homeassistant/components/lutron.py

```diff
@@ -110,3 +113,45 @@ def name(self):
     def should_poll(self):
         """No polling needed."""
         return False
+
+
+class LutronButton:
+    """Representation of a button on a Lutron keypad.
+
+    This is responsible for firing events as keypad buttons are pressed
+    (and possibly released, depending on the button type). It is not
+    represented as an entity; it simply fires events.
+    """
+
+    def __init__(self, hass, keypad, button):
+        """Register callback for activity on the button."""
+        self._name = '{}: {}'.format(keypad.name, button.name)
+        self._hass = hass
+        self._button = button
+        self._has_release_event = 'RaiseLower' in button.button_type
+        self._id = slugify(self._name)
+        self._event = 'lutron_{}'.format(CONF_EVENT)
+
+        button.subscribe(self.button_callback, None)
+
+    @callback
+    def button_callback(self, button, context, event, params):
+        """Fire an event about a button being pressed or released."""
+        from pylutron import Button
+
+        if self._has_release_event:
+            # A raise/lower button; we will get callbacks when the button is
+            # pressed and when it's released, so fire events for each.
+            if event == Button.Event.PRESSED:
+                action = 'pressed'
+            else:
+                action = 'released'
+        else:
+            # A single-action button; the Lutron controller won't tell us
+            # when the button is released, so use a different action name
+            # than for buttons where we expect a release event.
+            action = 'single'
+
+        data = {CONF_ID: self._id, CONF_EVENT: action}
+
+        self._hass.bus.fire(self._event, data, EventOrigin.remote)
```

Comment:
The origin is not remote, ie not from another home assistant instance. We can remove that argument.